<launch>
  <arg name="filename" default="/media/kazuhiro/Data/TIS_camera/2018-02-11/20_autoware.pcd"/>
  <arg name="interval" default="1"/>
  <node pkg="pcl_ros" type="pcd_to_pointcloud" name="pcd_to_pointcloud" args="$(arg filename) $(arg interval)" output="screen" />

  <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" />
  <!-- Run a passthrough filter to clean NaNs -->
  <node pkg="nodelet" type="nodelet" name="passthrough_x" args="load pcl/PassThrough pcl_manager" output="screen">
    <remap from="~input" to="/cloud_pcd" />
	<remap from="~output" to="/cloud_pcd_x" />
    <rosparam>
      filter_field_name: x
      filter_limit_min: 0.0
      filter_limit_max: 50.0
      filter_limit_negative: False
    </rosparam>
  </node>
  <node pkg="nodelet" type="nodelet" name="passthrough_xy" args="load pcl/PassThrough pcl_manager" output="screen">
    <remap from="~input" to="/cloud_pcd_x" />
	<remap from="~output" to="/cloud_pcd_xy" />
    <rosparam>
      filter_field_name: y
      filter_limit_min: 0.0
      filter_limit_max: 20.0
      filter_limit_negative: False
    </rosparam>
  </node>
  <node pkg="nodelet" type="nodelet" name="passthrough_xyz" args="load pcl/PassThrough pcl_manager" output="screen">
    <remap from="~input" to="/cloud_pcd_xy" />
	<remap from="~output" to="/cloud_pcd_xyz" />
    <rosparam>
      filter_field_name: z
      filter_limit_min: -1.0
      filter_limit_max: 50.0
      filter_limit_negative: False
    </rosparam>
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find pcl_common)/config/pcl_passthrough.rviz" />

  <node pkg="rqt_reconfigure" name="rqt_reconfigure" type="rqt_reconfigure" />
</launch>
